torchtitan/datasets/hf_datasets.py

```diff
@@ -72,47 +61,56 @@ def __init__(
         rank: int = 0,
         infinite: bool = False,
     ) -> None:
-        # allow user to pass in a (local or HF hub) path to use unsupported datasets
+        # Force lowercase for consistent comparison
+        dataset_name = dataset_name.lower()
+
         if dataset_name not in _supported_datasets:
-            if dataset_path:
-                logger.warning(
-                    f"Dataset {dataset_name} is not tested or verfied. "
-                    f"Recommended datasets are: {list(_supported_datasets.keys())}"
-                )
-            else:
-                raise ValueError(
-                    f"Dataset {dataset_name} is not supported. "
-                    f"Supported datasets are: {list(_supported_datasets.keys())}"
-                )
+            raise ValueError(
+                f"Dataset {dataset_name} is not supported. "
+                f"Supported datasets are: {list(_supported_datasets.keys())}"
+            )
 
         if not dataset_path:
             dataset_path = _supported_datasets[dataset_name]
-        logger.info(f"Preparing {dataset_name} dataset from {dataset_path}")
 
-        if dataset_name == "c4":
-            # c4 is huge, and requires both streaming and language selection
-            # (we default to en)
-            ds = load_dataset(dataset_path, name="en", split="train", streaming=True)
-        else:
-            ds = load_dataset(dataset_path, split="train")
+        if dataset_name not in DATASET_LOADERS:
+            raise ValueError(f"No loader found for dataset {dataset_name}")
```

Comment:
There are three such checks. I was thinking: can we merge the three dictionaries into one? With the value field being a tuple of `(path, load_method, text_processing_method`). This way one can't miss some of them, and we don't need the repeated checks here.